### DISCUSSION TITLE
Thoughts on Leetcode Premium?

### DISCUSSION THREAD

**Discussion by kanjurer** [jenndryden/Canadian-Tech-Internships-and-New-Grad-2025#152] [Polls]
Share your feelings

**Comment by jenndryden** [comment]
I think if you use LeetCode a lot it is a nice investment, especially during recruiting season months. You could even share an account with a friend to save money. However, you can definitely get by without it! Following the Neetcode 150 and Blind 75 are one of the best ways to practice for tech interviews, and Neetcode made a really great roadmap [here](https://neetcode.io/practice). When practicing for coding interviews, its definitely way better to understand fundementals of different kinds of problems (arrays, graphs, linked lists, trees, etc).. rather than randomly solving problems on Leetcode without a structure.

Another tool that personally helps me a lot is [LeetHub](https://engrmuhammadusman108.medium.com/leethub-a-browser-plugin-that-pushes-your-leetcode-solution-to-github-bb647ae40bc2). It connects your Leetcode submissions to Github automatically, so everytime you press "submit" on Leetcode it pushes a code file to your Github repo. Then you can easily access all of your Leetcode submissions in one place and really helps refresh your memory when you are practicing for interviews. 

I've also tried other interview prep methods, like [AlgoExpert](https://www.algoexpert.io/frontend/product). Personally, I think the algorithms was very similar to what already exists with LeetCode, but I really found it helpful preparing for Frontend/JS related interviews. If you have CSS/HTML interviews it has a built in compiler to see what the frontend actually builds like. 

In the end, you don't need to spend hundreds of dollars on interview prep. If you are good at finding resources on your own and self studying with Youtube videos, you can certainly get by without paying extra money. There are also lots of Discord servers and online sites that share what specific companies LeetCode premium problems are, so with some digging you can find that information for free. I think it is definitely is more convenient to have all of these things in one place, but certainly not necessary. 😊
